base/error.go

```diff
@@ -38,6 +39,22 @@ func ErrorAsHTTPStatus(err error) (int, string) {
 	if err == nil {
 		return 200, "OK"
 	}
+
+	switch err.Error() {
+	case gocbcore.ErrKeyNotFound.Error():
+		return http.StatusNotFound, "missing"
+	case gocbcore.ErrKeyExists.Error():
+		return http.StatusConflict, "Conflict"
+	case gocbcore.ErrTimeout.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrOverload.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrBusy.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrTmpFail.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	}
+
```

Comment:
It gets a bit convoluted taking the gocb hybrid bucket into consideration.  We can catch up offline on that, but in the meantime I filed a ticket to get rid of that hybrid bucket: https://github.com/couchbase/sync_gateway/issues/2423

After that's done, I think we'll be in a better position to loop back to this error handling and make it more sane.